spec/tapioca/compilers/dsl/active_record_associations_spec.rb

```diff
@@ -0,0 +1,317 @@
+# typed: false
+# frozen_string_literal: true
+
+require "spec_helper"
+
+describe("Tapioca::Compilers::Dsl::ActiveRecordAssociations") do
+  before(:each) do
+    require "tapioca/compilers/dsl/active_record_associations"
+  end
+
+  subject do
+    Tapioca::Compilers::Dsl::ActiveRecordAssociations.new
+  end
+
+  describe("#initialize") do
+    def constants_from(content)
+      with_content(content) do
+        subject.processable_constants.map(&:to_s).sort
+      end
+    end
+
+    it("gathers no constants if there are no ActiveRecord subclasses") do
+      assert_empty(subject.processable_constants)
+    end
+
+    it("gathers only ActiveRecord subclasses") do
+      content = <<~RUBY
+        class Post < ActiveRecord::Base
+        end
+
+        class Current
+        end
+      RUBY
+
+      assert_equal(constants_from(content), ["Post"])
+    end
+
+    it("rejects abstract ActiveRecord subclasses") do
```

Comment:
Can we also add a case where `Post < Something < ActiveRecord::Base`?

Reply:
I added it to this test case down below. I assumed that's what you were suggesting by commenting on this test